components/kyma-environment-broker/internal/process/deprovisioning/manager.go

```diff
@@ -72,13 +80,26 @@ func (m *Manager) Execute(operationID string) (time.Duration, error) {
 	}
 	operation := *op
 
+	logOperation := m.log.WithFields(logrus.Fields{"operation": operationID, "instanceID": operation.InstanceID})
+
 	provisioningOp, err := m.operationStorage.GetProvisioningOperationByInstanceID(op.InstanceID)
 	if err != nil {
 		m.log.Errorf("Cannot fetch ProvisioningOperation for instanceID %s from storage: %s", op.InstanceID, err)
+
```

Comment:
when err!=nil here when getting provisioningOperation, we also need to assign operation.State=domain.Failed, right? as here it gives up the deprovisioning. @wozniakjan

Reply:
does it give up deprovisioning or just errors and requeues? I think unless you mark the operation as failed explicitly, it will continue to requeue it until it times out.

Reply:
It returns `return 0, err` here which means it will not be requeued (code https://github.com/kyma-project/control-plane/blob/fc5da85e9872f49a5e73910733503ee549112029/components/kyma-environment-broker/internal/process/queue.go#L87). so in this failure, I think the operation.State needs to be set to Failed, or to consider to retry (requeue) it? (need to r`eturn when(non-zero), nil` to requeue/retry)

Reply:
I discussed this with the authors and it looks like you may have discovered a bug.

@piotrmiskiewicz mentioned that this place should probably check if the error from `GetProvisioningOperationByInstanceID()` is recoverable or not. If it's for example network issue, operation should be requeued, but if DB returns not found, then not much we can do but fail the operation explicitly like it's done in

https://github.com/kyma-project/control-plane/blob/fc5da85e9872f49a5e73910733503ee549112029/components/kyma-environment-broker/internal/process/provisioning/check_cluster_configuration.go#L68

Thanks for pointing this out, feel free to leave it as is for now, it's probably outside of the scope of your PR to fix all bugs :) I will create an issue to go through the requeueing logic, I have a feeling there might be more places like this in KEB.

Reply:
yes, this is better to raise a separated PR for this fix :) (it's out of this PR's purpose, just wanted to point it out:)). I will leave this publish event here, but just this event here will not have the failed state result.